tests/unit/cocoon/test_properties.py

```diff
@@ -373,3 +373,41 @@ def test_create_property_values(
         )
 
         self.assertEqual(first=property_values, second=expected_outcome)
+
+    @parameterized.expand([
+        [
+            "Just a code",
+            ["Strategy"],
+            "Traders",
+            "Transaction",
+            ["Transaction/Traders/Strategy"]
+        ],
+        [
+            "A scope and code",
+            ["Signal/Strategy"],
+            "Traders",
+            "Transaction",
+            ["Transaction/Signal/Strategy"]
+        ],
+        [
+            "A full key",
+            ["Portfolio/Signal/Strategy"],
+            "Traders",
+            "Transaction",
+            ["Portfolio/Signal/Strategy"]
+        ],
+    ])
+    def test_infer_full_property_keys(self, _, partial_keys, properties_scope, domain, expected_outcome):
```

Comment:
Just for completeness I would add `Transaction/Strategy` > `Transaction/Transaction/Strategy` to show that 2 part codes alway get interpreted as scope/code

Reply:
Done.